Further ZHA guidelines for Zigbee RSSI and LQI values (#33456)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

- In theory, an LQI value of 255 means a zero error rate, meaning it indicates absolute perfect link quality.
- In general, a positive high LQI value is better and a lower LQI value is worse. However, depending on your devices, that might not always be the reality.

> - Usually, if a device stays at an LQI value of 40+ or above then that link can usually be considered acceptable for normal operational conditions.

Reviewer: LQI isn't standard so I don't think it's useful to list exact numbers such as this. Furthermore, LQI is often a measure of the *last-hop* link quality, not of the connection between the device and the coordinator. Listing exact numbers leads people to try to mess with their network until the numbers get higher, often times without a problem existing to begin with.

Author: Removed numbers.

- In theory, an LQI value of 255 means a zero error rate, meaning it indicates absolute perfect link quality.
- In general, a positive high LQI value is better and a lower LQI value is worse. However, depending on your devices, that might not be the reality.

> - Usually, if a device stays at an LQI value of 40+ or above then that link can usually be considered acceptable for normal operational conditions.

Author: OK I have now committed something similar to that so will be closing this as resolved.

RSSI (Received Signal Strength Indicator) values are an indicator value of the raw signal strength between two devices. RSSI values are negative numbers in -dBm format (0 to -100 power ratio in decibels of the measured power referenced to one milliwatt). Lower negative values indicate less interference and a good signal. RSSI information is only between the endpoint device and the first hop from that device. As such, it may not necessarily show signal strength to the Zigbee Coordinator but instead could be showing signal strength to the nearest Zigbee Router device.

- Generally, anything -60 and above (meaning -50, -40, etc.) in RSSI should be considered a strong signal (not losing messages).
- Usually, anything at -80 and below (meaning -85, -90, etc.) in RSSI should be considered a noisy environment and you risk losing messages.

> LQI (Link Quality Index) values are shown as positive numbers on a scale but can be very hard to interpret for Zigbee and not as useful for troubleshooting. This is because the Zigbee specifications and the (IEEE 802.15.4 specification) do not standardize how to perform LQI measurements. The LQI value provided by the Zigbee devices is not measured using unified standards from all device manufacturers and Zigbee stacks, and often LQI only a measure of the last-hop link quality anyway which is most of the time not useful information, as such the values can not always be trusted. For example, Zigbee devices based on Silicon Labs EmberZNet stack use positive display numbers for LQI, where higher is better and lower is worse. The Texas Instruments Z-Stack computes LQI for each received packet from the raw “received signal strength index” (RSSI) by linearly scaling it between the minimum and maximum defined RF power levels for the radio that more or less just provides an LQI value that, based on the strength of the received signal.  This can be misleading in case the user has a noisy environment with interference within the same frequency range (as the RSSI value may be shown as increased even though the true link quality decreases). Other manufacturers and Zigbee stacks measure and calculate LQI values in another way. In theory, a positive high LQI value is better and a lower LQI value is worse, but depending on your devices, that might not always be the reality.

Reviewer 2: ```suggestion
LQI (Link Quality Index) values are shown as positive numbers on a scale but can be very hard to interpret for Zigbee and not as useful for troubleshooting. This is because the Zigbee specifications and the (IEEE 802.15.4 specification) do not standardize how to perform LQI measurements. The LQI value provided by the Zigbee devices is not measured using unified standards from all device manufacturers and Zigbee stacks, and often, LQI is only a measure of the last-hop link quality anyway, which is most of the time not useful information as such the values can not always be trusted.

For example, Zigbee devices based on Silicon Labs EmberZNet stack use positive display numbers for LQI, where higher is better and lower is worse. The Texas Instruments Z-Stack computes LQI for each received packet from the raw “received signal strength index” (RSSI) by linearly scaling it between the minimum and maximum defined RF power levels for the radio that more or less just provides an LQI value that, based on the strength of the received signal. This can be misleading in case you have a noisy environment with interference within the same frequency range (as the RSSI value may be shown as increased even though the true link quality decreases). Other manufacturers and Zigbee stacks measure and calculate LQI values in another way.

In theory, a positive high LQI value is better, and a lower LQI value is worse, but depending on your devices, that might not always be the reality.
```
